src/Tablebot/Plugins/Quote.hs

```diff
@@ -64,14 +76,41 @@ showQ qId m = do
       sendMessageVoid m $ pack $ txt ++ " - " ++ author
     Nothing -> sendMessageVoid m "Couldn't get that quote!"
 
+-- | @deleteQuote@, which looks for a message of the form @!quote delete n@,
+-- and removes it from the database.
+deleteQ :: Int64 -> Message -> DatabaseDiscord ()
+deleteQ qId m =
+  requirePermission Any m $
+    let k = toSqlKey qId
+     in do
+          qu <- get k
+          case qu of
+            Just (Quote _ _) -> do
+              delete k
+              sendMessageVoid m "Quote deleted"
+            Nothing -> sendMessageVoid m "Couldn't get that quote!"
+
 showQuoteHelp :: HelpPage
-showQuoteHelp = HelpPage "show" "show a quote by number" "**Show Quote**\nShows a quote by id\n\n*Usage:* `quote show <id>`" []
+showQuoteHelp = HelpPage "show" "show a quote by number" "**Show Quote**\nShows a quote by id\n\n*Usage:* `quote show <id>`" [] None
+
+deleteQuoteHelp :: HelpPage
+deleteQuoteHelp =
+  HelpPage
+    "delete"
+    "delete a quote by number"
+    [r|**Delete Quote**
+Delete a quote by id
+Requires moderation permission
+
+*Usage:* `quote delete <id>`|]
+    []
+    Any
```

Comment:
The one thing I dislike about this design (that I can't think of an immediate solution to) is the fact that help pages and their respective commands can have different permissions. I wonder if there's an alternative model where we have full subcommanding, each command is directly associated with a help page, and each command has a permission directly associated to it (so transitively the help page has it also). I think it may be something to consider when implementing subcommanding (which I think is the final blocking step after errors to having a "stable" plugin interface).